### ISSUE TITLE
Exported variables not recognised as exported, triggering SC2034

### ISSUE THREAD

**Issue by mrtimdog** [koalaman/shellcheck#958]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash -eux

declare -x a="a"
typeset -x b="b"

f()
{
    local -x c="c"
}

```

#### Here's what shellcheck currently says:

```

$ shellcheck myscript
 
Line 4:
typeset -x b="b"
           ^-- SC2034: b appears unused. Verify it or export it.
 
Line 8:
    local -x c="c"
             ^-- SC2034: c appears unused. Verify it or export it.

```

#### Here's what I wanted or expected to see:

No errors or warnings.

These are all exported variables, even the local one within the function scope, but they're only recognised as exported when using 'declare -x' or 'export', not with 'typeset -x' or 'local -x'.

**Comment by PX03AFK** [comment]
I have the same problem and even declaring them doesn't fix the issue.  In my case I am creating a number of variables within a for loop and shellcheck complains about them all.

**Comment by PX03AFK** [comment]
Sorry, should have been clear, I am exporting the variables within the loop.

**Comment by Nightfirecat** [comment]
Possible duplicate of #953?

**Comment by PX03AFK** [comment]
Looks as though it is a duplicate

**Comment by fauust** [comment]
Hi, same here (last version 0.6.0 installed with cabal).
`typeset -x`
should not give SC2034 warning.

**Comment by koalaman** [comment]
This is now fixed in 788aee1b7, thanks!
